### DISCUSSION TITLE
concourse_workers_tasks metric keeps reporting 0

### DISCUSSION THREAD

**Discussion by net-expertise** [concourse/concourse#8755] [Help & Support] [answered]
Hi,
I am trying to monitor the number of active tasks on each worker but it always returns 0, even when a job is running.
Other metrics like concourse_workers_containers are just working fine.

Port-forwarding Prometheus service 9090, here's what I get:
`concourse_workers_tasks{platform="linux",worker="concourse-worker-656496fff4-wvb7v"} 0
concourse_workers_tasks{platform="linux",worker="concourse-worker-656496fff4-tf6c4"} 0

concourse_workers_containers{platform="linux",tags="",team="",worker="concourse-worker-656496fff4-tf6c4"} 29
concourse_workers_containers{platform="linux",tags="",team="",worker="concourse-worker-656496fff4-wvb7v"} 5
`
I did not open an issue since it seems to be working for others.
Am I missing something?

**Comment by pn-santos** [comment, marked as answer]
If I'm not mistaken for that metric to work concourse needs to be run with the [limit-active-tasks-strategy](https://concourse-ci.org/container-placement.html#limit-active-tasks-strategy) enabled (don't remember if you also need to set `max-active-tasks-per-worker` too but I don't think so).

**Comment by net-expertise** [comment]
Indeed, I chained the limit-active-tasks to the volume-locality strategy with a high enough max tasks value
and I got this working
I'd like to use this metric to know when to scale down my deployments.

It'd be good to have this metric no matter the strategy IMHO
Thanks for the hint
